node/exts/erc20-bridge/signersvc/kwil.go

```diff
@@ -39,121 +46,336 @@ type EpochReward struct {
 }
 
-func (k *erc20rwExtApi) InstanceInfo(ctx context.Context) (*RewardInstanceInfo, error) {
-	procedure := "info"
-	input := []any{}
+func (k *signerClient) GetEpochRewards(ctx context.Context, namespace string, epochID *types.UUID) ([]*EpochReward, error) {
+	var rewards []*EpochReward
 
-	res, err := k.clt.Call(ctx, k.namespace, procedure, input)
-	if err != nil {
-		return nil, err
-	}
+	readTx := k.db.BeginDelayedReadTx()
+	defer readTx.Rollback(ctx)
 
-	if len(res.QueryResult.Values) == 0 {
-		return nil, nil
-	}
+	_, err := k.call.CallWithoutEngineCtx(ctx, readTx, namespace, "get_epoch_rewards", []any{epochID}, func(row *common.Row) error {
+		var ok bool
+		e := &EpochReward{}
+
+		e.Recipient, ok = row.Values[0].(string)
+		if !ok {
+			return fmt.Errorf("failed to get recipient")
+		}
+
+		e.Amount, ok = row.Values[1].(string)
+		if !ok {
+			return fmt.Errorf("failed to get amount")
+		}
+
+		rewards = append(rewards, e)
+
+		return nil
+	})
+
+	return rewards, err
+}
 
-	er := &RewardInstanceInfo{}
-	err = types.ScanTo(res.QueryResult.Values[0],
-		&er.Chain, &er.Escrow, &er.EpochPeriod, &er.Erc20, &er.Decimals, &er.Balance, &er.Synced, &er.SyncedAt, &er.Enabled)
+func (k *signerClient) VoteEpoch(ctx context.Context, namespace string, txSigner auth.Signer, epochID *types.UUID, safeNonce int64, signature []byte) (types.Hash, error) {
+	inputs := [][]any{{epochID, safeNonce, signature}}
+	res, err := k.execute(ctx, namespace, txSigner, "vote_epoch", inputs)
 	if err != nil {
-		return nil, err
+		return types.Hash{}, err
 	}
 
-	return er, nil
+	return res, nil
 }
 
-func (k *erc20rwExtApi) GetActiveEpochs(ctx context.Context) ([]*Epoch, error) {
-	procedure := "get_active_epochs"
-	input := []any{}
+func (k *signerClient) estimatePrice(ctx context.Context, tx *types.Transaction) (*big.Int, error) {
+	readTx := k.db.BeginDelayedReadTx()
+	defer readTx.Rollback(ctx)
 
-	res, err := k.clt.Call(ctx, k.namespace, procedure, input)
+	return k.kwilNode.Price(ctx, readTx, tx)
+
+}
+
+func (k *signerClient) accountNonce(ctx context.Context, acc *types.AccountID) (uint64, error) {
+	readTx := k.db.BeginDelayedReadTx()
+	defer readTx.Rollback(ctx)
+
+	_, nonce, err := k.kwilNode.AccountInfo(ctx, readTx, acc, true)
 	if err != nil {
-		return nil, err
+		return 0, fmt.Errorf("failed to get account info: %w", err)
 	}
 
-	if len(res.QueryResult.Values) == 0 {
-		return nil, nil
-	}
+	return uint64(nonce), nil
+}
 
-	ers := make([]*Epoch, len(res.QueryResult.Values))
-	for i, v := range res.QueryResult.Values {
-		er := &Epoch{}
-		err = types.ScanTo(v, &er.ID, &er.StartHeight, &er.StartTimestamp, &er.EndHeight,
-			&er.RewardRoot, &er.RewardAmount, &er.EndBlockHash, &er.Confirmed, &er.Voters, &er.VoteNonces, &er.VoteSignatures)
+// execute mimics client.Client.Execute, without client options.
+func (k *signerClient) execute(ctx context.Context, namespace string, txSigner auth.Signer, action string, tuples [][]any) (types.Hash, error) {
```

Comment:
still a bit hacky, if it's worth it we should implement a internal `TxSvcClient`. For now this is all I needed for signer service